backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -153,7 +153,12 @@ public Project getProjectById(String id, User user) throws SW360Exception {
         assertNotNull(project);
 
         if(!makePermission(project, user).isActionAllowed(RequestedAction.READ)) {
-            throw fail("User " + user + " is not allowed to view the requested project " + project + "!");
+            try {
+                fail("User " + user + " is not allowed to view the requested project " + project + "!");
+            } catch (SW360Exception exception) {
+                exception.setErrorCode(403);
```

Comment:
overload `fail` method by passing `error code` rather than catching and setting the `error code` in exception.

Reply:
Changes Made.